website/docs/guides/container_service_extension_install.html.markdown

```diff
@@ -320,9 +338,132 @@ The most common issues are:
+
+~> This section assumes that the old CSE v4.0 installation was done with Terraform by following the v4.0 guide steps.
+Also, you need to meet [the pre-requisites criteria](#pre-requisites).
+
+### Create the new RDE elements
+
+A new [RDE Interface][rde_interface] needs to be created. This one is completely new, introduced in version v4.1:
+
+```hcl
+resource "vcd_rde_interface" "cse_interface" {
+  vendor  = "cse"
+  nss     = "capvcd"
+  version = "1.0.0"
+  name    = "cseInterface"
+}
+```
+
+CSE v4.1 also requires the usage of [RDE Interface Behaviors][rde_interface_behavior] and
+[RDE Behavior Access Controls][rde_type_behavior_acl], that can be created with the following snippets (these can
+also be found in the [proposed configuration][step1]):
+
+```hcl
+resource "vcd_rde_interface_behavior" "capvcd_behavior" {
+  rde_interface_id = vcd_rde_interface.cse_interface.id
+  name             = "getFullEntity"
+  execution = {
+    "type" : "noop"
+    "id" : "getFullEntity"
+  }
+}
+
+resource "vcd_rde_type_behavior_acl" "capvcd_behavior_acl" {
+  rde_type_id      = vcd_rde_type.capvcdcluster_type_v120.id # This definition is below
+  behavior_id      = vcd_rde_interface_behavior.capvcd_behavior.id
+  access_level_ids = ["urn:vcloud:accessLevel:FullControl"]
+}
+```
+
+Create a new version of the [RDE Types][rde_type] that were used in v4.0. This will allow them to co-exist with the old ones,
+so we can perform a smooth upgrade.
+
+```hcl
+resource "vcd_rde_type" "vcdkeconfig_type_v110" {
+  # Same attributes as v4.1, except for:
+  version = "1.1.0" # New version
+  # New schema:
+  schema_url = "https://raw.githubusercontent.com/vmware/terraform-provider-vcd/main/examples/container-service-extension/v4.1/schemas/vcdkeconfig-type-schema-v1.1.0.json"
+}
+
+resource "vcd_rde_type" "capvcdcluster_type_v120" {
+  # Same attributes as v4.1, except for:
+  version = "1.2.0" # New version
+  # New schema:
+  schema_url = "https://raw.githubusercontent.com/vmware/terraform-provider-vcd/main/examples/container-service-extension/v4.1/schemas/capvcd-type-schema-v1.2.0.json"
+  # Notice that the new interface cse:capvcd:1.0.0 is used
+  interface_ids = [data.vcd_rde_interface.kubernetes_interface.id, vcd_rde_interface.cse_interface.id]
+  # Behaviors need to be created before any RDE Type
+  depends_on = [vcd_rde_interface_behavior.capvcd_behavior]
+}
+```
+
+### Upgrade the VCDKEConfig RDE (CSE Server configuration)
+
+With the new [RDE Types][rde_type] in place, you need to perform an upgrade of the existing `VCDKEConfig` [RDE][rde], which
+stores the CSE Server configuration. By using the v3.11.0 of the VCD Terraform Provider, you can do this update without forcing
+a replacement:
+
+```hcl
+resource "vcd_rde" "vcdkeconfig_instance" {
+  # Same values as before, except:
+  rde_type_id = vcd_rde_type.vcdkeconfig_type_v110.id # Update to the new RDE Type
+  input_entity = templatefile(var.vcdkeconfig_template_filepath, {
```

Comment:
nit
```suggestion
  rde_type_id  = vcd_rde_type.vcdkeconfig_type_v110.id # Update to the new RDE Type
  input_entity = templatefile(var.vcdkeconfig_template_filepath, {
```

Reply:
Good catch, I wonder why the `checkhcl` didn't catch this one 🤔